### PR TITLE
fix: event log arguments handling

### PR DESCRIPTION
**Pull request by ca11ab1e**
### What I did

fixes: #48.

Opening to simply provide a test case with expected values.

TODO:

- [x] fix filtering (https://github.com/ApeWorX/ape-starknet/pull/49/commits/8fb3428bdd100597801bc1d80a7bb5cc20038087)
- [x] fix uint256 (https://github.com/ApeWorX/ape-starknet/pull/49/commits/7abfda55e9a3f1aca3f13e1cb9842d545053ddcd)
- [x] ~fix ordering~ (fixed thanks to previous commits)

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [ ] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by ca11ab1e** [comment]
Good for review @unparalleled-js.

File: ape_starknet/ecosystems.py [hunk 1]
```diff
@@ -326,8 +326,26 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
         return txn_cls(**txn_data)
 
     def decode_logs(self, abi: EventABI, raw_logs: List[Dict]) -> Iterator[ContractLog]:
-        for index, log in enumerate(raw_logs):
-            event_args = dict(zip([a.name for a in abi.inputs], log["data"]))
+        event_key = get_selector_from_name(abi.name)
```
Comment (ca11ab1e) [inline ape_starknet/ecosystems.py:329]: https://starknet.io/docs/hello_starknet/events.html

File: ape_starknet/ecosystems.py [hunk 2]
```diff
@@ -326,8 +326,27 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
...
+        ) -> List[Union[int, Tuple[int, int]]]:
+            decoded: List[Union[int, Tuple[int, int]]] = []
+            iter_data = iter(data)
+            for abi_type in abi_types:
+                if abi_type.type == "Uint256":
+                    # unint256 are stored using 2 slots
+                    decoded.append((next(iter_data), next(iter_data)))
+                else:
+                    decoded.append(next(iter_data))
+            assert not list(iter_data)
```
Comment (unparalleled-js) [inline ape_starknet/ecosystems.py:343]: Let's keep assertions in the tests only.

Use an exception if needed, such as

```
if not list(iter_data):
    raise ValueError("Nothing to decode")
```

The asserts get stripped out of optimized python code and so I see them as mostly distracting and fills the code up with unnecessary lines.

**Review by unparalleled-js — APPROVED** [review]

File: ape_starknet/ecosystems.py (hunk 2 above)
Comment (ca11ab1e) [reply ape_starknet/ecosystems.py:343]: I think we can even remove it completely.

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**